start/software/container-how-to.md

```diff
@@ -0,0 +1,181 @@
+
+1. Editing the `Dockerfile`
+2. Editing the default image using local Docker
+
+We recommend the first option, as it is more reproducible, but the second option 
+can be useful for troubleshooting or especially tricky installs. 
+
+### Editing the `Dockerfile`
+
+Create a folder on your computer and inside it, create a blank text file 
+called `Dockerfile`.  
+
+The first line of this file should include the keyword `FROM` and then 
+the name of a Docker image (from Docker Hub) you want 
+to use as your starting point. If using the OSG's Ubuntu Xenial image that 
+would look like this: 
+
+	FROM opensciencegrid/osgvo-ubuntu-xenial
+
+Then, for each command you want to run to add libraries or software, use the 
+keyword `RUN` and then the command. Sometimes it makes sense to string 
+commands together using the `&&` operator and line breaks `\`, like so:
+
+	RUN apt-get update && \
+	    apt-get install -yy build-essentials
+
+or
+
+	RUN wget https://cran.r-project.org/src/base/R-3/R-3.6.0.tar.gz && \
+	    tar -xzf R-3.6.0.tar.gz && \
+	    cd R-3.6.0 && \
+	    ./configure && \
+	    make && \
+	    make install
+
+You can have multiple `RUN` commands for multiple installation steps.  
+
+(For all the possible Dockerfile keywords, see the [Docker Documentation](https://docs.docker.com/engine/reference/builder/))
+
+Once your Dockerfile is ready, you can "build" the container image by running this command: 
+
+    $ docker build -t namespace/repository_name .
+
+Note that the naming convention for Docker images is your Docker Hub username and then 
+a name you choose for that particular container image. So if my Docker Hub username 
+is `alice` and I created an image with the NCBI `blast` tool, I might use this name: 
+
+    $ docker build -t alice/NCBI-blast .
+
+
+### Editing the default image using local Docker
+
+You can also build an image interactively, without a Dockerfile. First, get 
+the desired starting image from Docker Hub. Again, we will
+look at the OSG Ubuntu Xenial image. 
+
+    $ docker pull opensciencegrid/osgvo-ubuntu-xenial
+
+We will run the image in a docker interactive session
+
+    $ docker run -it --name <docker_session_name_here> opensciencegrid/osgvo-ubuntu-xenial /bin/bash
+
+Giving the session a name is important because it will make it easier to 
+reattach the session later and commit the changes later on. Now you will 
+be greeted by a new command line prompt that will look something like this
+
+    [root@740b9db736a1 /]#
+
+You can now install the software that you need through the default package 
+manager, in this case `apt-get`. 
+
+    [root@740b9db736a1 /]# apt-get install build-essentials
+
+Once you have installed all the software, you simply `exit`
+
+    [root@740b9db736a1 /]# exit
+
+Now you can commit the changes to the image: 
+
+    docker commit <docker_session_name_here>
+
+Alternatively, you can use the sessions hash (`740b9db736a1` in the above example).
```

Comment:
Perhaps: "you can use the sessions hash as found in the command prompt (`740b9db736a1` in the above example)."